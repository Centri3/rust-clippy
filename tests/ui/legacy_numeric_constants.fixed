//@run-rustfix
//@aux-build:proc_macros.rs:proc-macro
#![allow(clippy::no_effect, deprecated, unused)]
#![warn(clippy::legacy_numeric_constants)]

#[macro_use]
extern crate proc_macros;

use std::u128 as _;
pub mod a {
    pub use std::u128;
}

fn main() {
    f32::EPSILON;
    u8::MIN;
    usize::MIN;
    u32::MAX;
    u32::MAX;
    use std::u32::MAX;
    u32::MAX;
    u32::MAX;
    u8::MAX;
    u8::MIN;
    ::std::primitive::u8::MIN;
    ::std::u8::MIN;
    ::std::primitive::u8::MIN;
    std::primitive::u32::MAX;
    u128::MAX;
    // Don't lint
    use std::f64;
    f32::EPSILON;
    u8::MIN;
    std::f32::consts::E;
    f64::consts::E;
    external! {
        ::std::primitive::u8::MIN;
        ::std::u8::MIN;
        ::std::primitive::u8::min_value();
    }
}

#[clippy::msrv = "1.42.0"]
fn msrv_too_low() {
    // FIXME: Why does this lint??
    u32::MAX;
}

#[clippy::msrv = "1.43.0"]
fn msrv_juust_right() {
    u32::MAX;
}
