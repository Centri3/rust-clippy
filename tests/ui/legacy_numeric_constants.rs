//@aux-build:proc_macros.rs
#![allow(clippy::no_effect, deprecated, unused)]
#![warn(clippy::legacy_numeric_constants)]

#[macro_use]
extern crate proc_macros;

use std::u128 as _;
pub mod a {
    pub use std::u128;
}

fn main() {
    std::f32::EPSILON;
    std::u8::MIN;
    std::usize::MIN;
    std::u32::MAX;
    core::u32::MAX;
    use std::u32;
    use std::u32::MAX;
    MAX;
    u32::MAX;
    u32::max_value();
    u8::max_value();
    u8::min_value();
    ::std::primitive::u8::MIN;
    ::std::u8::MIN;
    ::std::primitive::u8::min_value();
    std::primitive::u32::max_value();
    use std::f64;
    f64::MAX;
    self::a::u128::MAX;
    u128::MAX;
    // Don't lint
    f32::EPSILON;
    u8::MIN;
    std::f32::consts::E;
    use std::f32;
    f64::consts::E;
    external! {
        ::std::primitive::u8::MIN;
        ::std::u8::MIN;
        ::std::primitive::u8::min_value();
    }
}

#[clippy::msrv = "1.42.0"]
fn msrv_too_low() {
    // FIXME: Why does this lint??
    std::u32::MAX;
}

#[clippy::msrv = "1.43.0"]
fn msrv_juust_right() {
    std::u32::MAX;
}
