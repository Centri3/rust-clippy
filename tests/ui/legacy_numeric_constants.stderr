error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:15:5
   |
LL |     std::f32::EPSILON;
   |     ^^^^^^^^^^^^^^^^^ help: use the associated constant instead: `f32::EPSILON`
   |
   = note: `-D clippy::legacy-numeric-constants` implied by `-D warnings`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:16:5
   |
LL |     std::u8::MIN;
   |     ^^^^^^^^^^^^ help: use the associated constant instead: `u8::MIN`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:17:5
   |
LL |     std::usize::MIN;
   |     ^^^^^^^^^^^^^^^ help: use the associated constant instead: `usize::MIN`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:18:5
   |
LL |     std::u32::MAX;
   |     ^^^^^^^^^^^^^ help: use the associated constant instead: `u32::MAX`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:19:5
   |
LL |     core::u32::MAX;
   |     ^^^^^^^^^^^^^^ help: use the associated constant instead: `u32::MAX`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:21:5
   |
LL |     MAX;
   |     ^^^ help: use the associated constant instead: `u32::MAX`

error: usage of a legacy numeric method
  --> $DIR/legacy_numeric_constants.rs:22:10
   |
LL |     u32::max_value();
   |          ^^^^^^^^^^^ help: use the associated constant instead: `MAX`

error: usage of a legacy numeric method
  --> $DIR/legacy_numeric_constants.rs:23:9
   |
LL |     u8::max_value();
   |         ^^^^^^^^^^^ help: use the associated constant instead: `MAX`

error: usage of a legacy numeric method
  --> $DIR/legacy_numeric_constants.rs:24:9
   |
LL |     u8::min_value();
   |         ^^^^^^^^^^^ help: use the associated constant instead: `MIN`

error: usage of a legacy numeric method
  --> $DIR/legacy_numeric_constants.rs:27:27
   |
LL |     ::std::primitive::u8::min_value();
   |                           ^^^^^^^^^^^ help: use the associated constant instead: `MIN`

error: usage of a legacy numeric method
  --> $DIR/legacy_numeric_constants.rs:28:26
   |
LL |     std::primitive::u32::max_value();
   |                          ^^^^^^^^^^^ help: use the associated constant instead: `MAX`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:29:5
   |
LL |     self::a::u128::MAX;
   |     ^^^^^^^^^^^^^^^^^^ help: use the associated constant instead: `u128::MAX`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:46:5
   |
LL |     std::u32::MAX;
   |     ^^^^^^^^^^^^^ help: use the associated constant instead: `u32::MAX`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants.rs:51:5
   |
LL |     std::u32::MAX;
   |     ^^^^^^^^^^^^^ help: use the associated constant instead: `u32::MAX`

error: aborting due to 14 previous errors

